lib/utils/getSelectorMeaningfulSource.mjs

```diff
@@ -0,0 +1,45 @@
+import selectorParser from 'postcss-selector-parser';
+
+/**
+ * Get the source index of a selector, excluding leading whitespace.
+ *
+ * @param {import('postcss-selector-parser').Selector} node
+ *
+ * @returns {{selectorStr: string, index: number, endIndex: number}}
+ */
+export default function getSelectorMeaningfulSource(node) {
+	const firstIndex = node.nodes.findIndex((x) => !selectorParser.isComment(x));
+	const lastIndex = node.nodes.findLastIndex((x) => !selectorParser.isComment(x));
+
+	const first = node.nodes[firstIndex];
+	let last = node.nodes[lastIndex];
+
+	if (!first || !last) {
+		const selectorStr = node.toString().trim();
+		const index = node.sourceIndex ?? 0;
```

Comment:
`?? 0` when the AST node was constructed instead of being the result of parsing

Reply:
If this gets merged, I would expect a comment in the source.

Reply:
Yes, I will likely submit a patch for this but I am not looking forward to all the type errors everyone will get after that 😅